packages/accounts-controller/src/AccountsController.ts

```diff
@@ -247,12 +286,55 @@ export class AccountsController extends BaseController<
   }
 
   /**
-   * Returns the selected internal account.
+   * Returns the selected evm internal account by default unless the namespace is not eip155
    *
+   * @param chainId - Caip2 Id of the account
    * @returns The selected internal account.
    */
-  getSelectedAccount(): InternalAccount {
-    return this.getAccountExpect(this.state.internalAccounts.selectedAccount);
+  getSelectedAccount(
+    chainId: CaipChainId = ETHEREUM_CAIP_ID_WILDCARD,
+  ): InternalAccount {
+    if (!isCaipChainId(chainId) && chainId !== ETHEREUM_CAIP_ID_WILDCARD) {
```

Comment:
Maybe right now we should just throw an error if we use a CAIP2 chain ID? Like we should only consider namespaces since we use them against the `account.type`, WDYT?

Reply:
I think it should probably be like list accounts where it can be both a namespace or caip chain id.